### ISSUE TITLE
Enforce define on Param paths

### ISSUE THREAD

**Issue by chimon2000** [mrgnhnt96/go_router_paths#2]
One of the things that we have run into when using this library is that developers will occasionally forget to call `define` on paths that have a param in them.  Is there a way to make this logic a bit safer?  

It seems like it should be possible if `path` is removed from the `Param` class and `define` returns a `Path` object.

**Comment by mrgnhnt96** [comment]
Yeah, I've run into the same thing. There is an exception being thrown when `path` is called, which helps at runtime! But I agree, it'd be nice to have something alert you that you forgot about `define`.

I don't have time to work on this immediately. Though, I might be able to get to it within the coming weeks! If you have any suggestions or would like to fork the repo and create a PR, I'll welcome it!
